File: go/simple_calculator/simple_calculator_9.go
// simple_calculator.go
// This program is a simple calculator that performs basic arithmetic operations such as addition, subtraction, multiplication, and division.

package main

import (
    "fmt"
)

func main() {
    var num1, num2 float64
    var operator string

    fmt.Print("Enter first number: ")
    fmt.Scanln(&num1)
    fmt.Print("Enter second number: ")
    fmt.Scanln(&num2)
    fmt.Print("Enter operator (+, -, *, /): ")
    fmt.Scanln(&operator)

    var result float64
    switch operator {
    case "+":
        result = num1 + num2
    case "-":
        result = num1 - num2
    case "*":
        result = num1 * num2
    case "/":
        if num2 != 0 {
            result = num1 / num2
        } else {
            fmt.Println("Error: Division by zero")
            return
        }
    default:
        fmt.Println("Error: Invalid operator")
        return
    }

    fmt.Printf("Result: %.2f\n", result)
}

File: perl/simple_calculator/simple_calculator_9.pl
# This program is a simple calculator written in Perl. It performs basic arithmetic operations
# such as addition, subtraction, multiplication, and division. The user is prompted to enter
# two numbers and an operator, and the program outputs the result of the operation.

use strict;
use warnings;

print "Enter first number: ";
my $num1 = <STDIN>;
chomp($num1);

print "Enter second number: ";
my $num2 = <STDIN>;
chomp($num2);

print "Enter operator (+, -, *, /): ";
my $operator = <STDIN>;
chomp($operator);

my $result;

if ($operator eq '+') {
    $result = $num1 + $num2;
} elsif ($operator eq '-') {
    $result = $num1 - $num2;
} elsif ($operator eq '*') {
    $result = $num1 * $num2;
} elsif ($operator eq '/') {
    $result = $num1 / $num2;
} else {
    die "Invalid operator\n";
}

print "Result: $result\n";

